lib/core/versions/latest/AnchoredDataSerializer.ts

```diff
@@ -40,47 +35,34 @@ export default class AnchoredDataSerializer {
       throw new SidetreeError(ErrorCode.AnchoredDataIncorrectFormat, `Input is not in correct format: ${serializedData}`);
     }
 
-    const decodedNumberOfOperations = Encoder.decodeAsBuffer(splitData[0]);
-    const numberOfOperations = AnchoredDataSerializer.convertBufferToNumber(decodedNumberOfOperations);
+    const numberOfOperations = AnchoredDataSerializer.parsePositiveInteger(splitData[0]);
+
+    if (numberOfOperations > ProtocolParameters.maxOperationsPerBatch) {
+      throw new SidetreeError(
+        ErrorCode.AnchoredDataNumberOfOperationsGreaterThanMax,
+        `Number of operations ${numberOfOperations} must be less than or equal to ${ProtocolParameters.maxOperationsPerBatch}`
+      );
+    }
 
     return {
       anchorFileHash: splitData[1],
       numberOfOperations: numberOfOperations
     };
   }
 
-  private static convertNumberToBuffer (numberOfOperations: number): Buffer {
-
-    if (!Number.isInteger(numberOfOperations)) {
-      throw new SidetreeError(ErrorCode.AnchoredDataNumberOfOperationsNotInteger, `Number of operations ${numberOfOperations} must be an integer.`);
-    }
-
-    if (numberOfOperations < 0) {
-      throw new SidetreeError(ErrorCode.AnchoredDataNumberOfOperationsLessThanZero, `Number of operations ${numberOfOperations} must be greater than 0`);
-    }
-
-    if (numberOfOperations > this.maxUnsignedIntegerValue) {
-      // We are only using 4 bytes to store the number of operations so any number greater than
-      // that is not allowed.
-      throw new SidetreeError(ErrorCode.AnchoredDataNumberOfOperationsGreaterThanMax,
-                              `Number of operations ${numberOfOperations} must be less than equal to ${this.maxUnsignedIntegerValue}`);
-    }
-
-    // First write the input into a 4 bytes buffer. Little Endian format.
-    const byteArrayBuffer = Buffer.alloc(4);
-    byteArrayBuffer.writeUInt32LE(numberOfOperations, 0);
-
-    return byteArrayBuffer;
-  }
-
-  private static convertBufferToNumber (bytesBuffer: Buffer): number {
-
-    // Ensure that the input has 4 bytes
-    if (bytesBuffer.length !== 4) {
-      throw new SidetreeError(ErrorCode.AnchoredDataNumberOfOperationsNotFourBytes,
-                              `Input has ${bytesBuffer.length} bytes.`);
+  private static parsePositiveInteger (input: string): number {
+    // NOTE:
+    // /<expression>/ denotes regex.
+    // ^ denotes beginning of string.
+    // $ denotes end of string.
+    // [1-9] denotes leading '0' not allowed.
+    // \d* denotes followed by 0 or more decimal digits.
+    const isPositiveInteger = /^[1-9]\d*$/.test(input);

```

Comment:
Consider: may be possible to avoid regex and make it simpler?

````javascript
const number = Number(input);

if (Number.isNaN(number) || number < 0) throw;
````

Reply:
Will keep regex, `Number("0011")` is parsed as 11, which is not what we would want to allow.